l1-contracts/test/foundry/l1/unit/concrete/Utils/Utils.sol

```diff
@@ -125,6 +137,30 @@ library Utils {
         return logs;
     }
 
+    function createSystemLogsWithUpgradeTransactionForCTM(
+        bytes32 _expectedSystemContractUpgradeTxHash,
+        bytes32 _outputHash
+    ) public returns (bytes[] memory) {
+        bytes[] memory logsWithoutUpgradeTx = createSystemLogs(_outputHash);
+        bytes[] memory logs = new bytes[](logsWithoutUpgradeTx.length + 1);
+        for (uint256 i = 0; i < logsWithoutUpgradeTx.length; i++) {
+            logs[i] = logsWithoutUpgradeTx[i];
+        }
+        logs[2] = constructL2Log(
```

Comment:
probably makes sense to use `SystemLogKey.PREV_BATCH_HASH_KEY` as the index here to make it look less magical